### DISCUSSION TITLE
Materials: structure-property relationships and functional motifs

### DISCUSSION THREAD

**Discussion by rylandf** [Global-Chem/global-chem#110] [Ideas]
Hi everyone,
I've worked in a couple different disciplines but I'm participating here to help bridge the gap between small molecule chemistry and solid-state chemistry, hoping to find a way to extend GlobalChem to solid materials. I realize this is a little outside the expertise of many of the contributors here, but I hope that if I can present the problems in relatable terms then some of you may have a different perspective or knowledge base that helps us meet in the middle. 

To start with a relatively simple analogy consider molecular isomers, where you can have the same constituents but either connected differently or arranged differently in space. These isomers can have drastically different physical and chemical properties due to the structure of the molecule, which can affect everything from melting point to drug binding efficiency, optical properties, reactivity, solubility, etc. Now consider solid phase carbon. We are only talking about one element, and the carbon is only bound to other carbon, but we have such diverse materials as diamonds, graphite, graphene, carbon nanotubes, buckyballs, glassy carbon, and more exotic carbon allotropes. Diamond is one of the greatest thermal conductors known, but is a electrical insulator, while graphite is highly conductive within the individual layers, but acts as a semi-conductor moving from one sheet to the next. Diamonds are hard and are used as a paste to cut or polish things, while graphite is quite soft and often used as a dry lubricant because it slides across itself so easily. Graphene is essentially a single layer of graphite, and can even be separated from bulk graphite with simple tape, yet displays a number of properties not seen in bulk graphite. The number of layers of graphene and how they are twisted relative to each other can give exotic electron behavior and is it's own field of study within solid-state physics. 

To take things one step further, consider polymers. Polymers are materials made up of repeating monomer subunits, such as polyethylene being made up of ethylene units covalently bound to each other. However, the properties of polyethylene vary widely depending on factors not intrinsic to the polymer chains themselves. The length of the chains is one important factor, but chemically speaking the length of the chain being 1,000 units long or 1,000,000 is functionally identical. The difference in properties arises when you have a bunch of these chains together in a bulk system. The length of the chains and the amount of branching determine how densely the chains can be packed, which in turn determines how much surface area is available for intermolecular interactions. Many people are familiar with HDPE or LDPE from recycling goods at home: HDPE is used in goods such as milk jugs and garbage containers, while LDPE is used for things like plastic bags and packing materials. The differences in properties can be attributed to the number of monomer units in the chain, as well as processing conditions such as the temperature and pressure that the plastics are processed at, which affects the density and total intermolecular interactions.

One final example I will share is soda-lime silicate glass, which is what is used in most architectural, automotive, and beverage settings, is primarily composed of SiO2 tetrahedra cross-linked through the oxygen atoms. In cars, the front windshield is made out of glass processed in a different way than the windows on the sides or rear; side and rear window glass is tempered to make it scratch and chip resistant compared to the front windshield. This tempering is done (sometimes) by rapidly cooling the outside of the glass pane while the inside cools more slowly, which leads to a glass structure where the surface is in compression and the interior is in tension. This compression makes the surface harder to chip by containing surface fractures and preventing them from spreading. The stored energy in the glass due to tempering is why the glass shatters into small pieces when broken, compared to windows in your home which tend to create large shards. The large shards would be very dangerous in a car accident, which is why tempered glass is sometimes called safety glass. The front windshield is not made with safety glass because it is at much higher risk of debris from the road hitting it. If made out of tempered glass, a rock hitting the windshield at just the right angle would cause the entire windshield to shatter. To mitigate the risk of aforementioned shards of glass, the front windshield is instead covered in a laminate to contain the broken glass. Small rocks hitting the windshield will leave similarly small chips in the windshield which can be easily repaired if done before the cracks are allowed to propagate.

I say all of this to communicate that the processing conditions of the glass lead to different metastable configurations, and that these configurations lead to emergent properties which are not intrinsically related to the SiO2 tetrahedra of which the glass is comprised. This is a recurring theme in materials science, where materials are made up of common motifs (such as SiO2 tetrahedra or ethylene monomers), but many of the properties stem from macro scale organization of these motifs. To bring everything back to the beginning of this post, I see the following two tasks as necessary to apply GlobalChem to materials: (1) modelling subunits or motifs which can be tuned to reflect the local environment in a given material and (2) capturing the emergent, macroscale properties of materials which are made up of arrangements simple subunits. I believe the main value GlobalChem brings to the materials science world is that if flexible, representative forcefields could be created for common structural motifs then computational costs could be greatly reduced, or similarly system scales could be increased to better predict these emergent properties, while also creating a framework to examine how these motif properties relate to macroscale material properties.

High level overview - [The crucial role of functional motifs—microstructural units that govern material functions—in material research](https://phys.org/news/2022-04-crucial-role-functional-motifsmicrostructural-material.html)

Below are some papers I'm aware of on a similar idea of a fragment based formalism applied to solid systems:

- Dimensional Reduction: A Practical Formalism for Manipulating Solid Structures - DOI [10.1021/cm0007858](https://pubs.acs.org/doi/10.1021/cm0007858)
- Assessing Local Structure Motifs Using Order Parameters for Motif Recognition, Interstitial Identification, and Diffusion Path Characterization - DOI [10.3389/fmats.2017.00034](https://doi.org/10.3389/fmats.2017.00034)
- Fragment-Based Design of NbRuB as a New Metal-Rich Boride Superconductor - DOI [10.1021/cm504449s](https://doi.org/10.1021/cm504449s)
- Fragment formalism in main-group solids - DOI [10.1021/cm00007a004](https://doi.org/10.1021/cm00007a004)
- Material research from the viewpoint of functional motifs - DOI [10.1093/nsr/nwac017](https://doi.org/10.1093/nsr/nwac017)

**Comment by rylandf** [comment]
I know you like graphics Sul, I think there's some pretty clean structural graphics here. Strong inorganic chemists may like the last paper.
Also a good dissertation if anyone is looking to do a deep dive:
Toward the Systematic Design of Complex Materials from Structural Motifs - https://escholarship.org/uc/item/04g5p8ph

**Comment by Sulstice** [comment]
## Crystals

Life is usually bound by a set of rules and so we can refer to some previous proposed rules that will serve as precedence moving forward in the generation of the strings. Diamond and Graphite first:

 Let's refer to the openSMILES text, Page 25, Section 6.2, "Proposed Extensions for Polymers and Crystals": http://opensmiles.org/opensmiles.pdf 

<img width="712" alt="Screen Shot 2022-05-17 at 7 38 18 AM" src="https://user-images.githubusercontent.com/11812946/168802319-74c6fb54-a26d-46a8-b6da-1c5fc19fbe59.png">

`RDKit` will not support this and it might cause conflicts with internals we are not aware of, and I suspect a lot of other parsers might have the same. So here is what I propose, we install our own rules for processing crystal structures into SMILES sticking just to carbon for now and we can extend to others or some other graduate student.

The `GlobalChem` encoder will be a mapping like this, this will be only for GlobalChem and we can figure interoperability between software for different SMILES standards later. I want to avoid the SELFIES for now because I like to have control that is razor thin at this stage.

```python

    __SMILES_MAPPING__ = [
        ' ',
        '#', '%', '(', ')', '+', '-', '.', '/',
        '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
        '=', '@',
        'A', 'B', 'C', 'F', 'H', 'I', 'K', 'L', 'M', 'N', 'O', 'P',
        'R', 'S', 'T', 'V', 'X', 'Z',
        '[', '\\', ']',
        'a', 'b', 'c', 'e', 'g', 'i', 'l', 'n', 'o', 'p', 'r', 's',
        't', 'u',
        '&', ':', '*', '~',
    ]

```

I think that list of carbon crystals will be important and I will extend the encoder to match. GlobalChem will have an idea of what the crystals are.  The '~' comes from SMIRKS and one of the three family members I think that constructed it (SMILES, SMARTS, SMIRKS). It means virtual bond which comes from organic chemistry, where we used to draw these transition states with a dashed line. Think Sn2 Reactions:

<img width="492" alt="Screen Shot 2022-05-17 at 7 45 05 AM" src="https://user-images.githubusercontent.com/11812946/168803499-99852630-d20b-4705-b827-ac9205cc3e30.png">

The Br-----C------Cl is represented like this of `Br`  ~ `C` ~ `Cl`. We can account for different configurations this way, and convert it into `RDKit` since the compiler allows virtual bonds and virtual atoms to be processed without a formal atom type. 

Can you me get a list of common carbon-based materials?

## Polymers

"Many people are familiar with HDPE or LDPE from recycling goods at home: HDPE is used in goods such as milk jugs and garbage containers, while LDPE is used for things like plastic bags and packing materials. "

Yes you are right, and this reminds me of my youth "Macromolecular Chemistry" I have done the math for polymer production before and got a C, was too ballsy as a kid. It is tough. 

Plastics are actually a really good way to start is there a list you can make from that? There are different levels of densities. What we can do is actually algorithmically generate polymers of lengths of different densities as a feature. 

Read my paper on peptide generation: https://joss.theoj.org/papers/10.21105/joss.01992

I talk about automatic generation of peptide strings with lengths using "Middle Out Compression SMILES" is what I call it. I have to explain this one in more detail how it works but evidence of middle out compression is in GlobalChemExtensions already under different biological polymers:

RNA example:

```python

def create_sugar_backbone_layer(self):

        '''
        Creates the sugar backbone layer
        '''

        counter = 2

        start = '[*:1]C1CC(OP(=O)([O-])[O-])C'

        if self.sequence_length == 1:

            self.sugar_backbone = start + '(CO)O1'
            return self.smiles

        start = '[*:1]C1CC(OP(=O)([O-])[O-])C(COP(=O)([O-])'

        self.sugar_backbone += start

        for j in range(self.sequence_length - 1):

            self.sugar_backbone += 'OC2CC([*:%s])OC2COP(=O)([O-])' % counter
            counter += 1

        end = 'OC2CCOC2CO)O1'

        self.sugar_backbone += end

```
Something like this. See if you can understand what I am doing here.  Okay gotta pause I will continue more. Lots to go through.

**Comment by Sulstice** [comment]
@mitchell-travis87 What do you think about this idea about standardizing crystallographic data into Z-matrix internal coordinates. Since it's repeated pattern units. The official nomenclature was an ampersand. However, not agreed upon.

**Reply by ANUGAMAGE** [reply]
Hi @Sulstice 
I write this code using PySCF to create internal coordinates in colab for Diamond using your instructions.
I got an output but not sure whether it is write or wrong. 

```
pip install --prefer-binary pyscf
pip install --upgrade pyscf
pip install pyscf-forge
pip install pyscf[all]

from pyscf import gto
mol = gto.Mole()
mol.atom = '''
'diamond.xyz cordinates'
'''
mol.unit = 'AU'
mol.basis = 'sto-3g'
mol.build()
print(mol._atom)
print(mol.atom_coords(unit='ANG'))

```

Link : https://colab.research.google.com/drive/1xJssHrv26Fs1Lo0kgPznO4aac3H1aTXB?authuser=1#scrollTo=e4Tx1FtbDazj

**Comment by ANUGAMAGE** [comment]
> Hi everyone, I've worked in a couple different disciplines but I'm participating here to help bridge the gap between small molecule chemistry and solid-state chemistry, hoping to find a way to extend GlobalChem to solid materials. I realize this is a little outside the expertise of many of the contributors here, but I hope that if I can present the problems in relatable terms then some of you may have a different perspective or knowledge base that helps us meet in the middle.
> 
> To start with a relatively simple analogy consider molecular isomers, where you can have the same constituents but either connected differently or arranged differently in space. These isomers can have drastically different physical and chemical properties due to the structure of the molecule, which can affect everything from melting point to drug binding efficiency, optical properties, reactivity, solubility, etc. Now consider solid phase carbon. We are only talking about one element, and the carbon is only bound to other carbon, but we have such diverse materials as diamonds, graphite, graphene, carbon nanotubes, buckyballs, glassy carbon, and more exotic carbon allotropes. Diamond is one of the greatest thermal conductors known, but is a electrical insulator, while graphite is highly conductive within the individual layers, but acts as a semi-conductor moving from one sheet to the next. Diamonds are hard and are used as a paste to cut or polish things, while graphite is quite soft and often used as a dry lubricant because it slides across itself so easily. Graphene is essentially a single layer of graphite, and can even be separated from bulk graphite with simple tape, yet displays a number of properties not seen in bulk graphite. The number of layers of graphene and how they are twisted relative to each other can give exotic electron behavior and is it's own field of study within solid-state physics.
> 
> To take things one step further, consider polymers. Polymers are materials made up of repeating monomer subunits, such as polyethylene being made up of ethylene units covalently bound to each other. However, the properties of polyethylene vary widely depending on factors not intrinsic to the polymer chains themselves. The length of the chains is one important factor, but chemically speaking the length of the chain being 1,000 units long or 1,000,000 is functionally identical. The difference in properties arises when you have a bunch of these chains together in a bulk system. The length of the chains and the amount of branching determine how densely the chains can be packed, which in turn determines how much surface area is available for intermolecular interactions. Many people are familiar with HDPE or LDPE from recycling goods at home: HDPE is used in goods such as milk jugs and garbage containers, while LDPE is used for things like plastic bags and packing materials. The differences in properties can be attributed to the number of monomer units in the chain, as well as processing conditions such as the temperature and pressure that the plastics are processed at, which affects the density and total intermolecular interactions.
> 
> One final example I will share is soda-lime silicate glass, which is what is used in most architectural, automotive, and beverage settings, is primarily composed of SiO2 tetrahedra cross-linked through the oxygen atoms. In cars, the front windshield is made out of glass processed in a different way than the windows on the sides or rear; side and rear window glass is tempered to make it scratch and chip resistant compared to the front windshield. This tempering is done (sometimes) by rapidly cooling the outside of the glass pane while the inside cools more slowly, which leads to a glass structure where the surface is in compression and the interior is in tension. This compression makes the surface harder to chip by containing surface fractures and preventing them from spreading. The stored energy in the glass due to tempering is why the glass shatters into small pieces when broken, compared to windows in your home which tend to create large shards. The large shards would be very dangerous in a car accident, which is why tempered glass is sometimes called safety glass. The front windshield is not made with safety glass because it is at much higher risk of debris from the road hitting it. If made out of tempered glass, a rock hitting the windshield at just the right angle would cause the entire windshield to shatter. To mitigate the risk of aforementioned shards of glass, the front windshield is instead covered in a laminate to contain the broken glass. Small rocks hitting the windshield will leave similarly small chips in the windshield which can be easily repaired if done before the cracks are allowed to propagate.
> 
> I say all of this to communicate that the processing conditions of the glass lead to different metastable configurations, and that these configurations lead to emergent properties which are not intrinsically related to the SiO2 tetrahedra of which the glass is comprised. This is a recurring theme in materials science, where materials are made up of common motifs (such as SiO2 tetrahedra or ethylene monomers), but many of the properties stem from macro scale organization of these motifs. To bring everything back to the beginning of this post, I see the following two tasks as necessary to apply GlobalChem to materials: (1) modelling subunits or motifs which can be tuned to reflect the local environment in a given material and (2) capturing the emergent, macroscale properties of materials which are made up of arrangements simple subunits. I believe the main value GlobalChem brings to the materials science world is that if flexible, representative forcefields could be created for common structural motifs then computational costs could be greatly reduced, or similarly system scales could be increased to better predict these emergent properties, while also creating a framework to examine how these motif properties relate to macroscale material properties.
> 
> High level overview - [The crucial role of functional motifs—microstructural units that govern material functions—in material research](https://phys.org/news/2022-04-crucial-role-functional-motifsmicrostructural-material.html)
> 
> Below are some papers I'm aware of on a similar idea of a fragment based formalism applied to solid systems:
> 
> * Dimensional Reduction: A Practical Formalism for Manipulating Solid Structures - DOI [10.1021/cm0007858](https://pubs.acs.org/doi/10.1021/cm0007858)
> * Assessing Local Structure Motifs Using Order Parameters for Motif Recognition, Interstitial Identification, and Diffusion Path Characterization - DOI [10.3389/fmats.2017.00034](https://doi.org/10.3389/fmats.2017.00034)
> * Fragment-Based Design of NbRuB as a New Metal-Rich Boride Superconductor - DOI [10.1021/cm504449s](https://doi.org/10.1021/cm504449s)
> * Fragment formalism in main-group solids - DOI [10.1021/cm00007a004](https://doi.org/10.1021/cm00007a004)
> * Material research from the viewpoint of functional motifs - DOI [10.1093/nsr/nwac017](https://doi.org/10.1093/nsr/nwac017)

Hello @rylandf  My name is Anu, and I am from Sri Lanka. I am currently one of Sul's students in Global Chem, and I've been deeply fascinated by crystal structures since I was young. Growing up in a country abundant with unique gemstones like Serendibite and Ekanite, I often visited mining operations and analyzed various stone types. Over time, I developed a keen interest in understanding how gemstones behave under different lighting conditions and how their distinct shapes and colour are formed.

With a foundation in chemistry acquired through my education, after my graduation I became determined to delve deeper into the structures of gemstone materials. Recently, I've been thinking the use of AI to enhance the efficiency of gemstone mining operations and market analysis. Upon discussing my idea with Sul, he directed me towards this of material science discussion in Global-Chem.

In exploring this field further, I came across your discussions and delved into some of the research articles you referenced. It became evident to me that materials are far more complexed. I also gained an understanding of the significance of developing comprehensive and reliable databases for structures and properties of a vast array of materials.

Machine learning, as I learned, holds immense potential in predicting material properties efficiently and screening materials with desired properties. Moreover, the design of new materials hinges on the development of dependable databases and innovative methods for extracting functional motifs and structure-property relationships from machine learning models.

Your insights have been invaluable in shaping my understanding of this field, and I am really interested to explore further avenues in material science.
Thanks!
